### PR TITLE
.ci: Update CI scripts to make sure it works with Jenkins

### LINKED ISSUE #484: .ci: Modify CI scripts to make them work with Jenkins CI
We need to slightly modify the scripts in order to make sure everything is properly removed and that all the git branches are checked out on origin/master.

### PR DESCRIPTION
**Pull request by sboeuf**
Modify scripts to get Jenkins operational.

Fixes #484

### CODE REVIEW

**Comment by sboeuf** [comment]
@chavafg please take a look

File: .ci/lib.sh [hunk 1]
```diff
@@ -30,6 +30,10 @@ function clone_and_build() {
 
 	pushd ${project_dir}
 
+	# Ensure HEAD is at latest master.
```
Comment (devimc) [inline .ci/lib.sh]: emm nop, doing this you override the pull request code fetched by semaphore or any other CI

Comment (sboeuf) [reply .ci/lib.sh]: Ah yes you're right...

**Comment by sboeuf** [comment]
@devimc Here is the update

**Comment by chavafg** [comment]
lgtm

**Pull request merged**